### ISSUE TITLE
Can't read from S3 bucket after setting ACL='public-read' when creating the bucket.

### ISSUE THREAD

**Issue by anders33** [spulec/moto#1168]
Problem reading from s3 bucket after issue #1125 was solved.
This problem seems to be ralated to issue: Setting S3 bucket ACL has no effect #798

Test code:
```
 @mock_s3
 def test_s3_object_in_public_bucket():
    s3 = boto3.resource('s3')
    bucket = s3.Bucket('test-bucket')
    bucket.create(ACL='public-read')
    bucket.put_object(Body=b'ABCD', Key='file.txt')
    direct_url = 'https://test-bucket.s3.amazonaws.com/file.txt'
    response = requests.get(direct_url)
    assert response.status_code == 200
```

**Comment by JackDanger** [comment]
Thanks for finding this! Fixed in #1169

**Comment by JackDanger** [comment]
This is now fixed in Moto 1.1.12
